internal/http/services/owncloud/ocdav/put.go

```diff
@@ -172,32 +172,35 @@ func (s *svc) handlePut(ctx context.Context, w http.ResponseWriter, r *http.Requ
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
-		if tfRes.Status.Code != rpc.Code_CODE_OK {
-			log.Error().Interface("status", tfRes.Status).Msg("error touching file")
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
-		sRes, err := client.Stat(ctx, &provider.StatRequest{
-			Ref: ref,
-		})
-		if err != nil {
-			log.Error().Err(err).Msg("error sending grpc touch file request")
-			w.WriteHeader(http.StatusInternalServerError)
+		if tfRes.Status.Code == rpc.Code_CODE_OK {
+			sRes, err := client.Stat(ctx, &provider.StatRequest{
+				Ref: ref,
+			})
+			if err != nil {
+				log.Error().Err(err).Msg("error sending grpc touch file request")
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+			if sRes.Status.Code != rpc.Code_CODE_OK {
+				log.Error().Interface("status", sRes.Status).Msg("error touching file")
+				w.WriteHeader(http.StatusInternalServerError)
```

Comment:
urgh ... there are so many places where we should return a proper error using `HandleErrorStatus` and not just dumbly fall back to internal server error and I hate to bother you with this, but can you try to use `HandleErrorStatus` here?

Reply:
Thanks. Changed.